pkg/backend/mock.go

```diff
@@ -519,3 +519,38 @@ func (ms *MockStack) DefaultSecretManager(info *workspace.ProjectStack) (secrets
 	}
 	panic("not implemented")
 }
+
+//
+// Mock stack reference
+//
+
+// MockStackReference is a mock implementation of [StackReference].
+// Set the fields on this struct to control the behavior of the mock.
+type MockStackReference struct {
```

Comment:
There's possibly a few places around that were already making up mock stack ref types, might be worth doing a hunt to clean those up if my memory is right and they do exist.